patches/minecraft/net/minecraft/block/BlockCactus.java.patch

```diff
@@ -9,7 +9,27 @@
  {
      public static final PropertyInteger field_176587_a = PropertyInteger.func_177719_a("age", 0, 15);
      protected static final AxisAlignedBB field_185593_b = new AxisAlignedBB(0.0625D, 0.0D, 0.0625D, 0.9375D, 0.9375D, 0.9375D);
-@@ -110,8 +110,8 @@
+@@ -48,7 +48,9 @@
+             if (i < 3)
+             {
+                 int j = ((Integer)p_180650_3_.func_177229_b(field_176587_a)).intValue();
+-
++                net.minecraftforge.event.world.BlockEvent ev = new net.minecraftforge.event.world.BlockEvent.CropGrowEvent.Pre(p_180650_1_,p_180650_2_,p_180650_3_);
++                net.minecraftforge.common.MinecraftForge.EVENT_BUS.post(ev);
++                if(ev.getResult() != net.minecraftforge.fml.common.eventhandler.Event.Result.DENY) {
```

Comment:
This is where you would use ForgeHooks having this big block of code in each patch over and over is bad.